### ISSUE TITLE
LLM Feature Request: penalties

### ISSUE THREAD

**Issue by bgoelTT** [tenstorrent/tt-metal#32212]
The `frequency_penalty` and `presence_penalty` parameters in the OpenAI chat completions API https://platform.openai.com/docs/api-reference/chat/create#chat_create-frequency_penalty and https://platform.openai.com/docs/api-reference/chat/create#chat_create-presence_penalty respectively, signal the model to apply penalties (rescale logits) depending on the token history. This includes prompt & output tokens. 

There is also the `repetition_penalty` which is **not** a standard OpenAI API penalty, but is a vLLM one. According to https://docs.vllm.ai/en/latest/api/vllm/#vllm.SamplingParams it:
> Penalizes new tokens based on whether they appear in the prompt and the generated text so far. Values > 1 encourage the model to use new tokens, while values < 1 encourage the model to repeat tokens.

Therefore, the difference between `presence_penalty` & `repetition_penalty` is the tokens under consideration:
* `presence_penalty` applies to the **output, generated tokens**
* `repetition_penalty` applies to the **prompt and output, generated tokens**

We need to enable the configuration of these parameters in our on-device sampling implementation, for each request in a batch.

This issue serves as the parent issue for all tasks which are needed to expose this parameter.

**Comment by sraizada-tt** [comment]
status:
metal PR under review, verified that all ops with sub_core_grids work, only op still falling back is ttnn.scatter_add
vllm PR under review

**Comment by ekuslysTT** [comment]
Notes
- if k not set correctly causes greedy sampling --> fix on branch

Action Items
- rerun model CI, passing locally

**Comment by ekuslysTT** [comment]
@ppetrovicTT open new sub-issue for issue surfaced after Nikola's changes. @bgoelTT for vis

**Comment by ekuslysTT** [comment]
See updates on https://github.com/tenstorrent/tt-metal/issues/34090

**Comment by ekuslysTT** [comment]
Closing after 12/31 release
